code/modules/events/grid_check.dm

```diff
@@ -45,6 +45,8 @@
 			continue
 		if(C.cell)
 			C.cell.charge = 0
+	for(var/obj/machinery/power/P in GLOB.machines)
+		P.malfunction = TRUE
```

Comment:
Shouldn't this check if the machine is in one of the protected areas and skip them? It also seems to not check for the z level like the previous check, so wouldn't this also affect gateway missions and so on?

Reply:
oh that would probably be important lol

Reply:
Did some code diving, this is definitely an issue. If machinery no longer produces power during a grid check, then...the engine doesn't produce any power, meaning it can't power its own containment. The apc can still bridge it usually since it is exempt from the check, but it could run out, couldn't it?

Reply:
added the checks

Reply:
now i need another check